openmls/src/messages/external_proposals.rs

```diff
@@ -41,12 +41,12 @@ impl JoinProposal {
     /// * `epoch` - group's epoch
     /// * `signer` - of the sender to sign the message
     #[allow(clippy::new_ret_no_self)]
-    pub fn new<Storage: StorageProvider>(
+    pub fn new<Provider: OpenMlsProvider>(
```

Comment:
Why does this have to be a full OpenMLSProvider now?

Reply:
It makes calling it easier. But you're right, it's not needed. I reverted the change. You see in the tests what this may require to call it.